File: src/main/java/io/github/oleksivio/telegram/bot/core/model/method/CaptionAction.java
package io.github.oleksivio.telegram.bot.core.model.method;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;

public abstract class CaptionAction extends FormattedMessageAction {

    /**
     * caption String Optional File caption (may also be used when resending file by file_id),
     * 0-200 characters
     */
    @JsonProperty("caption")
    private String caption = null;

    protected CaptionAction(String method, ActionNetworker actionNetworker) {
        super(method, actionNetworker);
    }

    protected void pSetCaption(String caption) {
        this.caption = caption;
    }

    public String getCaption() {
        return caption;
    }

    public abstract CaptionAction setCaption(String caption);
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/model/method/CaptionAction.kt
package io.github.oleksivio.telegram.bot.core.model.method

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker

abstract class CaptionAction protected
constructor(method: String, actionNetworker: ActionNetworker)
    : FormattedMessageAction(method, actionNetworker) {

    /**
     * caption String Optional File caption (may also be used when resending file by file_id),
     * 0-200 characters
     */
    @JsonProperty("caption")
    var caption: String? = null

}
